src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.RemoveNewKeywordAction.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.HideBase
+{
+    internal partial class HideBaseCodeFixProvider
+    {
+        private class RemoveNewKeywordAction : CodeActions.CodeAction
+        {
+            private readonly Document _document;
+            private readonly SyntaxNode _node;
+
+            public override string Title => CSharpFeaturesResources.Remove_new_keyword;
+
+            public RemoveNewKeywordAction(Document document, SyntaxNode node)
+            {
+                _document = document;
+                _node = node;
+            }
+
+            protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
+            {
+                var syntaxFacts = CSharpSyntaxFacts.Instance;
+
+                var newModifier = GetNewModifier(_node);
+                Debug.Assert(newModifier != default, $"'new' keyword was not found, but diagnostic was triggered");
+
+                var newNode = _node;
+
+                if (newModifier.HasTrailingTrivia || newModifier.HasLeadingTrivia)
```

Comment:
one of these will always be true afaict.

Reply:
Here's an example of a `new` modifier with neither true:

```cs
class C
{
new(int a, int b) x;
}
```

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzhgHwAEAmAAgGEBYAKAG9aA7GAdwAoBLRjM7NMrj2ABKMggDctAL60gA

Reply:
Thanks, interesting example, will add as a test case (together with others like inner class, indexer, etc from another comment after agreeing on updating `HideBaseCodeFixProvider` to support more members in addition to property/method/field).